reana_server/fetcher.py

```diff
@@ -295,16 +297,24 @@ class WorkflowFetcherZip(WorkflowFetcherBase):
     """Fetch the specification of a workflow from a zip archive."""
 
     def __init__(
-        self, parsed_url: ParsedUrl, output_dir: str, spec: Optional[str] = None
+        self,
+        parsed_url: ParsedUrl,
+        output_dir: str,
+        spec: Optional[str] = None,
+        workflow_name: Optional[str] = None,
     ):
         """Initialize the workflow specification fetcher.
 
         :param parsed_url: Parsed URL of the workflow specification to fetch.
         :param output_dir: Directory where all the data will be saved to.
         :param spec: Optional path to the workflow specification.
+        :param workflow_name: Workflow name that overrides the workflow name generation.
         """
         super().__init__(parsed_url, output_dir, spec)
         self._archive_name = self._parsed_url.basename
+        self._workflow_name = None
+        if workflow_name:
+            self._workflow_name = self._clean_workflow_name(workflow_name)
```

Comment:
*suggestion:* To have cleaning of workflow name in one place maybe it is better to move this part to `generate_workflow_name` method.

```python
if workflow_name:
    return self._clean_workflow_name(workflow_name)
return self._clean_workflow_name(self._parsed_url.basename_without_extension)
```

Or even better, in `init`:

```python
if workflow_name:
    self._workflow_name = self._clean_workflow_name(workflow_name)
else:
    self._workflow_name = self._clean_workflow_name(self._parsed_url.basename_without_extension)
```